File: SpaceGuildBack/c#/location.cs
using System;
using System.Collections.Generic;
using Microsoft.AspNetCore.Components.Routing;

namespace SpaceGuildBack;

public abstract class Location : canMove
{
    public List<Location> Links { get; set; } = new List<Location>();
    public List<string> PlayerTokens { get; set; } = new List<string>();

    public void AddLink(Location location)
    {
        Links.Add(location);
    }

    public void RemoveLink(Location location)
    {
        Links.Remove(location);
    }

    public void ReceiveToken(string token)
    {
        PlayerTokens.Add(token);
    }

    public void Move(string token, Location destination)
    {
        if (Links.Contains(destination) && PlayerTokens.Contains(token))
        {
            PlayerTokens.Remove(token);
            destination.ReceiveToken(token);
        }
        
    }
}


public class Orbit : Location
{
    
}

public class Station : Location
{
    
}

File: SpaceGuildBack/python/location.py


class Location(canMove):
    def __init__(self):
        self.links:Location = []
        # self.player_tokens:str = []

    def add_link(self, location):
        self.links.append(location)

    def remove_link(self, location):
        if location in self.links:
            self.links.remove(location)

    # def receive_token(self, token):
    #     self.player_tokens.append(token)

    # def move(self, token, destination):
    #     if destination in self.links and token in self.player_tokens:
    #         self.player_tokens.remove(token)
    #         destination.receive_token(token)





class Orbit(Location,canScan,canAttack):
    pass

class Station(Location,canRepair,canMarket):
    pass

class Resource(Location,canGather,canScan):
    pass





class canScan:
    pass

class canGather:
    @staticmethod
    def gather():
        pass

class canAttack:
    @staticmethod
    def attack():
        pass

class canRepair:
    @staticmethod
    def repair():
        pass

class canMissions:
    def missions(self):
        print("printing missions!")

class canMarket:
    @staticmethod
    def market():
        pass

